### PR TITLE
feat(components): [progress] support set striped progress bar

### PR DESCRIPTION
**Pull request by wzc520pyfm**
Set the striped progress bar with the `striped` property.

Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/progress/src/progress.ts [hunk 1]
```diff
@@ -57,6 +57,10 @@ export const progressProps = buildProps({
     ]),
     default: '',
   },
+  striped: {
+    type: Boolean,
+    default: false,
```
Comment (btea) [inline packages/components/progress/src/progress.ts:62]: `default: false` can be removed.

File: packages/theme-chalk/src/progress.scss [hunk 2]
```diff
@@ -120,6 +120,11 @@
       transform: translateZ(0);
       animation: indeterminate 3s infinite;
     }
+
+    @include m(striped) {
+      background-image: linear-gradient(45deg, rgba(0, 0, 0, 0.1) 25%, transparent 25%, transparent 50%, rgba(0, 0, 0, 0.1) 50%, rgba(0, 0, 0, 0.1) 75%, transparent 75%, transparent);
+      background-size: 1.25rem 1.25rem;
```
Comment (btea) [inline packages/theme-chalk/src/progress.scss:126]: It might be better to use `em` here.

**Review by ryuhangyeong — APPROVED** [review]

**Review by btea — APPROVED** [review]
LGTM

**Comment by btea** [comment]
Maybe can add a property to control whether to add animation, what do you think? @wzc520pyfm @ryuhangyeong

**Comment by wzc520pyfm** [comment]
> Maybe can add a property to control whether to add animation, what do you think? @wzc520pyfm @ryuhangyeong

I think it's cool, we can add a `striped-flow` property to get the stripes to flow? let's me try.

**Comment by ryuhangyeong** [comment]
@btea A very cool idea!

File: docs/examples/progress/striped-progress.vue [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    <el-progress
+      :percentage="percentage"
+      :stroke-width="15"
+      status="exception"
+      striped
+      striped-flow
+      :duration="duration"
+    />
+    <el-button @click="percentage -= 10">-</el-button>
+    <el-button @click="percentage += 10">+</el-button>
```
Comment (btea) [inline docs/examples/progress/striped-progress.vue:28]: You can use `el-button-group`, consistent with other examples.

File: packages/theme-chalk/src/progress.scss [hunk 4]
```diff
@@ -120,6 +120,15 @@
...
     }
+
+    @include m(striped) {
+      background-image: linear-gradient(45deg, rgba(0, 0, 0, 0.1) 25%, transparent 25%, transparent 50%, rgba(0, 0, 0, 0.1) 50%, rgba(0, 0, 0, 0.1) 75%, transparent 75%, transparent);
+      background-size: 1.25em 1.25em;
+    }
+
+    @include m(striped-flow) {
+      animation: striped-flow 3s linear infinite;
+    }
```
Comment (btea) [inline packages/theme-chalk/src/progress.scss:131]: `striped-flow` seems to take effect after adding `striped`, maybe the styles here should be moved inside `striped`？

Comment (wzc520pyfm) [reply packages/theme-chalk/src/progress.scss:131]: Do you mean that a component gets a flowing striped after you add a `striped` property to it? but, 
I add a `striped` property and all I get is a striped that doesn't flow. A striped appear after a `striped` is added. After a `striped` and a `striped-flow` are added, the striped flows.

This is the sample code:
https://element-plus.run/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IGNvbXB1dGVkLCByZWYsIHZlcnNpb24gYXMgdnVlVmVyc2lvbiB9IGZyb20gJ3Z1ZSdcbmltcG9ydCB7IHZlcnNpb24gYXMgRXBWZXJzaW9uIH0gZnJvbSAnZWxlbWVudC1wbHVzJ1xuaW1wb3J0IHsgTWludXMsIFBsdXMgfSBmcm9tICdAZWxlbWVudC1wbHVzL2ljb25zLXZ1ZSdcbiAgXG5jb25zdCBwZXJjZW50YWdlID0gcmVmPG51bWJlcj4oNzApXG5jb25zdCBkdXJhdGlvbiA9IGNvbXB1dGVkKCgpID0+IE1hdGguZmxvb3IocGVyY2VudGFnZS52YWx1ZSAvIDEwKSlcblxuY29uc3QgaW5jcmVhc2UgPSAoKSA9PiB7XG4gIHBlcmNlbnRhZ2UudmFsdWUgKz0gMTBcbiAgaWYgKHBlcmNlbnRhZ2UudmFsdWUgPiAxMDApIHtcbiAgICBwZXJjZW50YWdlLnZhbHVlID0gMTAwXG4gIH1cbn1cbmNvbnN0IGRlY3JlYXNlID0gKCkgPT4ge1xuICBwZXJjZW50YWdlLnZhbHVlIC09IDEwXG4gIGlmIChwZXJjZW50YWdlLnZhbHVlIDwgMCkge1xuICAgIHBlcmNlbnRhZ2UudmFsdWUgPSAwXG4gIH1cbn1cbjwvc2NyaXB0PlxuXG48dGVtcGxhdGU+XG4gIDxkaXYgc3R5bGU9XCJ3aWR0aDogMzUwcHg7XCI+XG4gICAgPGVsLXByb2dyZXNzIDpwZXJjZW50YWdlPVwiNTBcIiA6c3Ryb2tlLXdpZHRoPVwiMTVcIiBzdHJpcGVkIC8+XG4gICAgPGVsLXByb2dyZXNzXG4gICAgICA6cGVyY2VudGFnZT1cIjMwXCJcbiAgICAgIDpzdHJva2Utd2lkdGg9XCIxNVwiXG4gICAgICBzdGF0dXM9XCJ3YXJuaW5nXCJcbiAgICAgIHN0cmlwZWRcbiAgICAgIHN0cmlwZWQtZmxvd1xuICAgIC8+XG4gICAgPGVsLXByb2dyZXNzXG4gICAgICA6cGVyY2VudGFnZT1cIjEwMFwiXG4gICAgICA6c3Ryb2tlLXdpZHRoPVwiMTVcIlxuICAgICAgc3RhdHVzPVwic3VjY2Vzc1wiXG4gICAgICBzdHJpcGVkXG4gICAgICBzdHJpcGVkLWZsb3dcbiAgICAgIDpkdXJhdGlvbj1cIjEwXCJcbiAgICAvPlxuICAgIDxlbC1wcm9ncmVzc1xuICAgICAgOnBlcmNlbnRhZ2U9XCJwZXJjZW50YWdlXCJcbiAgICAgIDpzdHJva2Utd2lkdGg9XCIxNVwiXG4gICAgICBzdGF0dXM9XCJleGNlcHRpb25cIlxuICAgICAgc3RyaXBlZFxuICAgICAgc3RyaXBlZC1mbG93XG4gICAgICA6ZHVyYXRpb249XCJkdXJhdGlvblwiXG4gICAgLz5cbiAgICA8ZWwtYnV0dG9uLWdyb3VwPlxuICAgICAgPGVsLWJ1dHRvbiA6aWNvbj1cIk1pbnVzXCIgQGNsaWNrPVwiZGVjcmVhc2VcIiAvPlxuICAgICAgPGVsLWJ1dHRvbiA6aWNvbj1cIlBsdXNcIiBAY2xpY2s9XCJpbmNyZWFzZVwiIC8+XG4gICAgPC9lbC1idXR0b24tZ3JvdXA+XG4gIDwvZGl2PlxuPC90ZW1wbGF0ZT4iLCJQbGF5Z3JvdW5kTWFpbi52dWUiOiI8c2NyaXB0IHNldHVwPlxuaW1wb3J0IEFwcCBmcm9tICcuL0FwcC52dWUnXG5pbXBvcnQgeyBzZXR1cEVsZW1lbnRQbHVzIH0gZnJvbSAnLi9lbGVtZW50LXBsdXMuanMnXG5zZXR1cEVsZW1lbnRQbHVzKClcbjwvc2NyaXB0PlxuXG48dGVtcGxhdGU+XG4gIDxBcHAgLz5cbjwvdGVtcGxhdGU+XG4iLCJpbXBvcnRfbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7XG4gICAgXCJlbGVtZW50LXBsdXNcIjogXCJodHRwczovL3ByZXZpZXctMTIzMzMtZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9pbmRleC5mdWxsLm1pbi5tanNcIixcbiAgICBcImVsZW1lbnQtcGx1cy9cIjogXCJ1bnN1cHBvcnRlZFwiXG4gIH1cbn0iLCJpbXBvcnQtbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7XG4gICAgXCJ2dWVcIjogXCJodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL0B2dWUvcnVudGltZS1kb21AbGF0ZXN0L2Rpc3QvcnVudGltZS1kb20uZXNtLWJyb3dzZXIuanNcIixcbiAgICBcIkB2dWUvc2hhcmVkXCI6IFwiaHR0cHM6Ly9mYXN0bHkuanNkZWxpdnIubmV0L25wbS9AdnVlL3NoYXJlZEBsYXRlc3QvZGlzdC9zaGFyZWQuZXNtLWJ1bmRsZXIuanNcIixcbiAgICBcImVsZW1lbnQtcGx1c1wiOiBcImh0dHBzOi8vcHJldmlldy0xMjMzMy1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL2luZGV4LmZ1bGwubWluLm1qc1wiLFxuICAgIFwiZWxlbWVudC1wbHVzL1wiOiBcInVuc3VwcG9ydGVkXCIsXG4gICAgXCJAZWxlbWVudC1wbHVzL2ljb25zLXZ1ZVwiOiBcImh0dHBzOi8vZmFzdGx5LmpzZGVsaXZyLm5ldC9ucG0vQGVsZW1lbnQtcGx1cy9pY29ucy12dWVAMi9kaXN0L2luZGV4Lm1pbi5qc1wiXG4gIH0sXG4gIFwic2NvcGVzXCI6IHt9XG59IiwiZWxlbWVudC1wbHVzLmpzIjoiaW1wb3J0IHsgZ2V0Q3VycmVudEluc3RhbmNlIH0gZnJvbSAndnVlJ1xuaW1wb3J0IEVsZW1lbnRQbHVzIGZyb20gJ2VsZW1lbnQtcGx1cydcblxubGV0IGluc3RhbGxlZCA9IGZhbHNlXG5hd2FpdCBsb2FkU3R5bGUoKVxuXG5leHBvcnQgZnVuY3Rpb24gc2V0dXBFbGVtZW50UGx1cygpIHtcbiAgaWYgKGluc3RhbGxlZCkgcmV0dXJuXG4gIGNvbnN0IGluc3RhbmNlID0gZ2V0Q3VycmVudEluc3RhbmNlKClcbiAgaW5zdGFuY2UuYXBwQ29udGV4dC5hcHAudXNlKEVsZW1lbnRQbHVzKVxuICBpbnN0YWxsZWQgPSB0cnVlXG59XG5cbmV4cG9ydCBmdW5jdGlvbiBsb2FkU3R5bGUoKSB7XG4gIHJldHVybiBuZXcgUHJvbWlzZSgocmVzb2x2ZSwgcmVqZWN0KSA9PiB7XG4gICAgY29uc3QgbGluayA9IGRvY3VtZW50LmNyZWF0ZUVsZW1lbnQoJ2xpbmsnKVxuICAgIGxpbmsucmVsID0gJ3N0eWxlc2hlZXQnXG4gICAgbGluay5ocmVmID0gJ2h0dHBzOi8vcHJldmlldy0xMjMzMy1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL2luZGV4LmNzcydcbiAgICBsaW5rLmFkZEV2ZW50TGlzdGVuZXIoJ2xvYWQnLCByZXNvbHZlKVxuICAgIGxpbmsuYWRkRXZlbnRMaXN0ZW5lcignZXJyb3InLCByZWplY3QpXG4gICAgZG9jdW1lbnQuYm9keS5hcHBlbmQobGluaylcbiAgfSlcbn0iLCJfbyI6eyJzaG93SGlkZGVuIjp0cnVlLCJzdHlsZVNvdXJjZSI6Imh0dHBzOi8vcHJldmlldy0xMjMzMy1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL2luZGV4LmNzcyJ9fQ==

Comment (btea) [reply packages/theme-chalk/src/progress.scss:131]: I mean if I only added the `striped-flow` attribute without adding the `striped` attribute, the animation should not have to be added at this time.

Comment (btea) [reply packages/theme-chalk/src/progress.scss:131]: Like this, move the following code to line 127
```scss
&.#{$namespace}-progress-bar__inner--striped-flow {
  animation: striped-flow 3s linear infinite;
}
```

Comment (wzc520pyfm) [reply packages/theme-chalk/src/progress.scss:130]: Thanks! I understand. I have completed.

**Comment by btea** [comment]
Thank you for your contribution ❤️

**Pull request merged**